include/boost/redis/connection.hpp

```diff
@@ -298,6 +300,85 @@ struct reader_op {
    }
 };
 
+template <class Executor>
+struct health_checker_op {
+   connection_impl<Executor>* conn_;
+   asio::coroutine coro_{};
+
+public:
+   health_checker_op(connection_impl<Executor>& conn) noexcept
+   : conn_{&conn}
+   { }
+
+   template <class Self>
+   void operator()(Self& self, system::error_code ec = {}, std::size_t = {})
+   {
+      BOOST_ASIO_CORO_REENTER(coro_)
+      {
+         if (conn_->cfg_.health_check_interval == std::chrono::seconds::zero()) {
+            conn_->logger_.trace("ping_op (1): timeout disabled.");
+            BOOST_ASIO_CORO_YIELD asio::async_immediate(self.get_io_executor(), std::move(self));
+            self.complete(system::error_code{});
+            return;
+         }
+
+         for (;;) {
+            // Clean up any previous leftover
+            clear_response(conn_->ping_resp_);
+
+            // Execute the request
+            BOOST_ASIO_CORO_YIELD
+            {
+               auto* conn = conn_;  // avoid use-after-move problems
+               auto timeout = 2 * conn->cfg_.health_check_interval;
```

Comment:
The `2 *` factor should be removed now to [match the docs](https://github.com/boostorg/redis/blob/f955dc01d2c5b9695f347992c0136321a8ff5276/include/boost/redis/config.hpp#L107-L109).